src/tricks/steps/environment.rs

```diff
@@ -0,0 +1,117 @@
+
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::process::Command;
+use std::process::Stdio;
+use anyhow::{Context as _, Result};
+
+use super::RunStep;
+
+use crate::{
+    tricks::status::Status,
+};
+
+/// Spawn a container using the docker api.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(crate) struct CreateEnvironment {
+
+    pub kernel_tag: String,
+
+    pub kconfig: String,
+
+    pub bconfig: String,
+
+    /// Status on failure. Default is SetupFailure.
+    #[serde(default = "crate::serde_defaults::default_setup_failure")]
+    pub failure: Status,
+    /// Status on success. Default is Undecided.
+    #[serde(default)]
+    pub success: Status,
+}
+
+#[async_trait]
+impl RunStep for CreateEnvironment {
+    async fn do_run(&self) -> Result<()> {
+        let bconfig = String::from(&self.bconfig);
+        let kconfig = String::from(&self.kconfig);
+        create_buildroot_image(bconfig,kconfig);
+        launch_image();
+        Ok(())
+    }
+
+    fn on_success(&self) -> Status {
+        self.success
+    }
+
+    fn on_failure(&self) -> Status {
+        self.failure
+    }
+}
+
+fn create_buildroot_image(bconfig: String, kconfig: String){
+    let buildroot_folder = String::from("~/Desktop/buildroot-bpfcontain/buildroot");
+
+    let mut buildroot_config = String::from("BR2_DEFCONFIG=");
+    buildroot_config.push_str(&bconfig);
+
+    let mut kernel_config = String::from("BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE=");
+    kernel_config.push_str(&kconfig);
+
+    let test_cmd = String::from("make");
+    let mut test_args = Vec::new();
+    test_args.push(String::from("-C"));
+    test_args.push(buildroot_folder);
+    test_args.push(buildroot_config);
+    test_args.push(kernel_config);
+
+    run_environment_command(test_cmd, test_args);
+
+
+}
+
+fn launch_image(){
+    let test_cmd = String::from("qemu-system-x86_64");
+    let out = Command::new(&test_cmd)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .arg("-M")
+        .arg("pc")
+        .arg("-m")
+        .arg("2048")
+        .arg("-nographic")
+        .arg("-smp")
+        .arg("1")
+        .arg("-kernel")
+        .arg("/home/kevin/Desktop/buildroot-bpfcontain/buildroot/output/images/bzImage")
```

Comment:
Not portable. We need a generic path to a bzImage

Reply:
For the generic path I would say we want to use the `guest` command line option I mentioned earlier. But rather than a flag let's make this a separate hidden subcommand. If you want I can take care of this part, just let me know here.